compendium/DeclarativeServices/src/manager/ComponentConfigurationImpl.cpp

```diff
@@ -161,8 +161,14 @@ namespace cppmicroservices
         void
         ComponentConfigurationImpl::SetRegistrationProperties()
         {
-            if (regManager)
+            if (regManager && !currentlySettingProperties)
             {
+                currentlySettingProperties = true;
+                detail::ScopeGuard sg(
+                    [this]()
+                    {
+                        currentlySettingProperties = false;
+                    });
```

Comment:
a comment about why this block of code exists would be helpful for future maintainers. This code block solves a specific problem that isn't obvious from the code alone.